staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -772,8 +773,8 @@ type unstructuredSchemaValidator struct {
 	dropInvalidMetadata bool
 	validator           *validate.SchemaValidator
 	// empty or a top-level field name like "status" the validator should be applied to
-	validatedField string
-	defaulting     bool
+	validatedField      string
+	defaulting, pruning bool
 }
 
 func (v *unstructuredSchemaValidator) apply(u *unstructured.Unstructured) error {
```

Comment:
For those of us following along at home.  When does this get called?  In a flow that goes 
 1. decode to internal (or whatever that is for CR)
 2. call mutating admission
     1. convert to external for webhook
     2. convert back to internal
     3. repeat for all webhooks
 3. call rest strategy
 4. call prepareforOperation
 5. call validating admission
 6. call validate
 7. convert to storage version
 8. serialize to storage

I'd expect to have values pruned before mutating admission, after webhook patches are applied, and before validating admission, just .

If you don't do it before mutating admission, the first mutating webhook may get extraneous values.  If you don't do it after mutating admission and before validating admission, validation admission webhooks may get bad values.  If you don't do it before storage, a bad converter might cause you store bad values.